generator/src/main/java/com/scottlogic/deg/generator/generation/grouped/FieldSpecGroupValueGenerator.java

```diff
@@ -85,7 +86,7 @@ private FieldSpec updateFirstSpecFromRelations(Field first, FieldSpecGroup group
             .map(relation -> createModifierForField(first, relation, group))
             .map(Optional::of)
             .reduce(Optional.of(firstFieldSpec), this::mergeOptionalFieldspecs)
-            .orElseThrow(() -> new IllegalStateException("Failed to merge field specs in related fields"));
+            .orElseThrow(() ->  new ValidationException("The provided profile is wholly contradictory!"));
```

Comment:
I'm not sure I like the exclamation mark, but that's a stylistic thing.

Reply:
It's what was in the other comment, so consistency I guess?